x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SSLService.java

```diff
@@ -329,27 +341,27 @@ public SSLEngine createSSLEngine(SSLConfiguration configuration, String host, in
      *
      * @param sslConfiguration the configuration to check
      */
-    public boolean isConfigurationValidForServerUsage(SSLConfiguration sslConfiguration) {
-        Objects.requireNonNull(sslConfiguration, "SSLConfiguration cannot be null");
-        return sslConfiguration.keyConfig() != KeyConfig.NONE;
+    public boolean isConfigurationValidForServerUsage(SslConfiguration sslConfiguration) {
+        Objects.requireNonNull(sslConfiguration, "SslConfiguration cannot be null");
+        return sslConfiguration.getKeyConfig().hasKeyMaterial();
```

Comment:
Rather than have a special `NONE` instance like X-Pack did, and making code check for it, ssl-config has a real method instead.